scene/main/canvas_layer.cpp

```diff
@@ -258,13 +258,11 @@ void CanvasLayer::_update_follow_viewport(bool p_force_exit) {
 	}
 }
 
-#ifdef TOOLS_ENABLED
```

Comment:
I wonder if those were kept within `TOOLS_ENABLED` for a good reason. I think `_validate_property` is also used at runtime for some objects whose properties actually need validation, and not just editor-specific tweaks.

Reply:
I was the one who did that in the first place. :stuck_out_tongue_closed_eyes: I'm just removing it because all the other instances of this function (and many others that are editor-only) aren't encapsulated in `TOOLS_ENABLED`. There should be a PR dedicated only to tackle this issue if it ever happens.